src/sources/csv.rs

```diff
@@ -0,0 +1,261 @@
+//! CSV source for syncing with Famedly's Zitadel.
+
+use std::{fs, path::PathBuf};
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use csv::Reader;
+use serde::Deserialize;
+
+use super::Source;
+use crate::{user::User, zitadel::SourceDiff};
+
+/// CSV Source
+pub struct CsvSource {
+	/// CSV Source configuration
+	csv_config: CsvSourceConfig,
+}
+
+#[async_trait]
+impl Source for CsvSource {
+	fn get_name(&self) -> &'static str {
+		"CSV"
+	}
+
+	async fn get_diff(&self) -> Result<SourceDiff> {
+		let new_users = self.read_csv()?;
+		return Ok(SourceDiff { new_users, changed_users: vec![], deleted_user_ids: vec![] });
```

Comment:
Continuing on [the past discussion](https://github.com/famedly/ldap-sync/pull/49#discussion_r1754790510), I think we should at least add a comment that this only implements adding users.

Potentially this could work for changing users as well, as for deleting it would require different csv table format

Reply:
Agreed. I think we should also double check with marketing whether this is actually the *desired* behavior. Could well be that they didn't consider the edge case of "some existing user on an instance matches with the ones we want to sync but has outdated data".

Of course, if we want to update users we need to first query Zitadel, which is basically #53, and in fact updating users will make #53 significantly easier to make generic. We can merge this PR either way, but we should add a TODO if we want to change the behavior.

Reply:
I added an explanatory comment.